src/search.coffee

```diff
@@ -6,234 +6,259 @@ url = require 'url'
-    reqOptions
-
-  # Given a list of strings, generates a string wrapped in single quotes with
-  # the list entries separated by a `+`.
-  quoted: (values) ->
-    values = [values] unless _.isArray values
-    values = (v.replace "'", "''" for v in values)
-    "'#{values.join '+'}'"
-
-  # Generates a sequence of numbers no larger than the page size which the sum
-  # of the list equal to numResults.
-  generateTops: (numResults, pageSize = Search.PAGE_SIZE) ->
-    tops = [numResults % pageSize] if numResults % pageSize isnt 0
-    tops or= []
-    (pageSize for i in [0...Math.floor(numResults / pageSize)]).concat tops
-
-  # Generate a sequence of offsets as a multiple of page size starting at
-  # skipStart and ending before skipStart + numResults.
-  generateSkips: (numResults, skipStart) ->
-    skips = [skipStart]
-    for count in @generateTops(numResults)[...-1]
-      skips.push skips[skips.length - 1] + count
-    skips
-
-  parallelSearch: (vertical, options, callback) ->
-    opts = _.extend {}, {top: Search.PAGE_SIZE, skip: 0}, options
-
-    # Generate search options for each of the search requests.
-    pairs = _.zip @generateTops(opts.top), @generateSkips(opts.top, opts.skip)
-    requestOptions = _.map pairs, ([top, skip]) ->
-      _.defaults {top, skip}, options
-
-    search = (options, callback) =>
-      @search vertical, options, callback
-
-    async.mapLimit requestOptions, @parallel, search, callback
-
-  search: (vertical, options, callback) ->
-    requestOptions =
-      uri: "#{BING_SEARCH_ENDPOINT}/#{vertical}"
-      qs: _.extend @requestOptions(options),
-        $format: 'json'
-      auth:
-        user: @accountKey
-        pass: @accountKey
+  ###
+  The PAGE_SIZE variable is used for `count` on most API requests. The
+  documentation states that the maximum results is API specific; but, all of
+  the API references just state that "the actual number delivered may be less
+  than requested." From what I've seen, searches typically return ~30  results;
+  however, incomplete result sets occur at any `count > 10`. `25` is a good
+  number for avoiding holes in data and duplicate results across pages.
+  ###
+  @PAGE_SIZE = 25
+
+  ###
+  The MAX_RESULTS variable, used for the legacy `top` pagination option, allows
+  for us to return the same default `50` results as v1.0.1 of this library.
+  ###
+  @MAX_RESULTS = 50
+
+  ###
+  We only need to one result (zero is impossible) from most verticals. With
+  a web search, however, the `totalEstimatedMatches` needs to be checked
+  from a higher page for accurate data.
+
+  The 1,000 value comes from empirical data. It seems that after 600
+  results, the accuracy gets quite consistent and accurate. I picked 1,000
+  just to be in the clear. It also doesn't matter if there are fewer than
+  1,000 results.
+  ###
+  @COUNT_ACCURACY_OFFSET = 1000
+
+  constructor: (@accountKey, @parallelLimit = 10, @useGzip = true) ->
```

Comment:
I saw you had an early commit making `@accountKeys` and was able to find this in the documentation; but, as far as I could tell, we only needed one key for the 4 "primary" APIs (web/image/video/news). While testing with one key, I only received an authorization error when I tried to access the [Autosuggest](https://msdn.microsoft.com/en-us/library/mt711406.aspx)/[Speller](https://msdn.microsoft.com/en-us/library/mt711413.aspx) APIs which are currently not implemented.

Just re-confirmed this, since it's definitely an odd/unexpected behavior:
```
sam@sam-work ~ $ curl -I -X GET https://api.cognitive.microsoft.com/bing/v5.0/search?q=Moz -H "Ocp-Apim-Subscription-Key: [redacted]"
HTTP/1.1 200 OK
Content-Length: 10730

sam@sam-work ~ $ curl -I -X GET https://api.cognitive.microsoft.com/bing/v5.0/images/search?q=Moz -H "Ocp-Apim-Subscription-Key: [redacted]"
HTTP/1.1 200 OK
Content-Length: 99683

sam@sam-work ~ $ curl -I -X GET https://api.cognitive.microsoft.com/bing/v5.0/videos/search?q=Moz -H "Ocp-Apim-Subscription-Key: [redacted]"
HTTP/1.1 200 OK
Content-Length: 65182

sam@sam-work ~ $ curl -I -X GET https://api.cognitive.microsoft.com/bing/v5.0/news/search?q=Moz -H "Ocp-Apim-Subscription-Key: [redacted]"
HTTP/1.1 200 OK
Content-Length: 12477

sam@sam-work ~ $ curl -I -X GET https://api.cognitive.microsoft.com/bing/v5.0/Suggestions?q=Moz -H "Ocp-Apim-Subscription-Key: [redacted]"
HTTP/1.1 401 Access Denied
Content-Length: 143

sam@sam-work ~ $ curl -I -X GET https://api.cognitive.microsoft.com/bing/v5.0/SpellCheck?q=Moz -H "Ocp-Apim-Subscription-Key: [redacted]"
HTTP/1.1 401 Access Denied
Content-Length: 143
```